localstack/services/apigateway/provider.py

```diff
@@ -492,11 +494,10 @@ def update_method(
                 continue
 
             if path == "/authorizationType" and value in ("CUSTOM", "COGNITO_USER_POOLS"):
```

Comment:
so it either update the type or the id that points to, correct?

Reply:
Yes, you can:
- Update the type, but if the type is either CUSTOM or COGNITO_USER_POOLS, you also need to provide a valid `AuthorizerId` in the call (so you'd need to provide both). 
- Update only the ID, it will be taken into account only if the existing authorisation type is CUSTOM or COGNITO_USER_POOLS

I can see there's one scenario missing here, if the type is already `CUSTOM` and we change it to `COGNITO_USER_POOLS`, technically I think we'd still need to update the `AuthorizerId`, but this hasn't been tested against AWS.

The validation logic is these `Update*` operation is pretty convoluted, and it takes a lot of testing to get all the possible combinations.